libr/anal/type.c

```diff
@@ -122,18 +122,119 @@ R_API void r_anal_save_parsed_type(RAnal *anal, const char *parsed) {
 static int typecmp(const void *a, const void *b) {
 	char *type1 = (char *) a;
 	char *type2 = (char *) b;
-	return strcmp (type1 , type2);
+	return strcmp (type1, type2);
 }
 
 R_API RList *r_anal_types_from_fcn(RAnal *anal, RAnalFunction *fcn) {
 	RListIter *iter;
 	RAnalVar *var;
 	RList *list = r_anal_var_all_list (anal, fcn);
-        RList *type_used = r_list_new ();
-        r_list_foreach (list, iter, var) {
-		r_list_append (type_used , var->type);
-        }
-	RList *uniq = r_list_uniq (type_used , typecmp);
+	RList *type_used = r_list_new ();
+	r_list_foreach (list, iter, var) {
+		r_list_append (type_used, var->type);
+	}
+	RList *uniq = r_list_uniq (type_used, typecmp);
 	r_list_free (type_used);
 	return uniq;
 }
+
+static void enum_type_fini(void *e, void *user) {
+	(void)user;
+	RAnalEnumCase *cas = e;
+	free ((char *)cas->name);
+}
+
+static RAnalBaseType *get_enum_type(RAnal *anal, const char *sanitized_name) {
+	r_return_val_if_fail (sanitized_name != NULL, NULL);
+
+	RAnalBaseType *base_type = malloc (sizeof (RAnalBaseType));
```

Comment:
use R_NEW0()

Reply:
What are the benefits of such macros instead of callocing directly?

Reply:
They are a bit shorter and more expressive. If you don't want to pre-initialize with 0, you can also use `R_NEW()`.